### ISSUE TITLE
Serial devices /dev/ttyS* having wrong ownership and permissions

### ISSUE THREAD

**Issue by mc-hof** [Microsoft/WSL#3042]
Windows build number:  Windows 10 Version 10.0.16299.309

The issue: Serial terminal Emulators like picocom refuse to work as access is denied to the serial port device (/dev/ttyS3 in my case).

I am not 100% sure wether this is an issue of WSL or of the Ubuntu app running on top of WSL.
The cause of the problem is, that all serial devices are owned by root and also belong to group root, instead to group dialout as is normally the case on any Linux Distribution that I know of.

Besides that, permissions of /dev/ttyS* are set to 640 in WSL. It should be set to 660 though (in addition to fix ownership as said above).

picocom works as expected if I change /dev/ttyS3 such that it belongs to group dialout and change permissions to 660, like this:

akfho@palm:~$ ls -l /dev/ttyS3
crw-rw---- 1 root dialout 4, 67 Mar 19 23:42 /dev/ttyS3

**Comment by benhillis** [comment]
@mc-hof - Good catch. Looks like we got the default mode / GID wrong. I'll get this fixed.

**Comment by mc-hof** [comment]
Thank you!

**Comment by mc-hof** [comment]
Ok. I don't use the Insider build but anyway thanks a lot for Fixing this. For the time being I can work around this issue.

**Comment by mobluse** [comment]
A problem is that Ubuntu for WSL from Store doesn't remember the changes, but you have to do them every time you start Ubuntu for WSL.

```
pi@lenovo5:~$ ls -l /dev/ttyS3
crw-r----- 1 root root 4, 67 Mar 30 10:00 /dev/ttyS3
pi@lenovo5:~$ sudo chgrp dialout /dev/ttyS3
[sudo] password for pi:
pi@lenovo5:~$ sudo chmod 660 /dev/ttyS3
pi@lenovo5:~$ ls -l /dev/ttyS3
crw-rw---- 1 root dialout 4, 67 Mar 30 10:00 /dev/ttyS3
```

**Comment by mc-hof** [comment]
Yes I know. But I don't shut down Ubunt on WSL often. I can wait for the fixed WSL release.

**Comment by sunilmut** [comment]
Fixed in [17639](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-17639-skip-ahead)

**Comment by ChuckM** [comment]
Awesome, now to wait for them to roll out to the regular release so that my company can install them on everyone's machine. 

Given the nature of this sort of problem, is there a feature request here for a script that runs as root just after WSL is initialized and launched and just prior to the prompt being displayed for the user? If such a capability existed then  anyone could have edited that file to add 'chmod 660 /dev/ttyS*' to it and it would have fixed this immediately. I can imagine other things that will pop up where such a mechanism will provide a reasonable path when people differ on what they thing the defaults should be.
